service/lib/agama/software/callbacks/signature.rb

```diff
@@ -50,37 +38,39 @@ def setup
           Yast::Pkg.CallbackImportGpgKey(
             Yast::FunRef.new(method(:import_gpg_key), "boolean (map <string, any>, integer)")
           )
+          Yast::Pkg.CallbackAcceptUnknownGpgKey(
+            Yast::FunRef.new(
+              method(:accept_unknown_gpg_key), "boolean (string, string, integer)"
+            )
+          )
+          Yast::Pkg.CallbackAcceptVerificationFailed(
+            Yast::FunRef.new(
+              method(:accept_verification_failed), "boolean (string, map <string, any>, integer)"
+            )
+          )
         end
 
         # Callback to handle unsigned files
         #
         # @param filename [String] File name
         # @param repo_id [Integer] Repository ID. It might be -1 if there is not an associated repo.
         def accept_unsigned_file(filename, repo_id)
-          repo = Yast::Pkg.SourceGeneralData(repo_id)
-          source = if repo
-            format(
-              _("The file %{filename} from repository %{repo_name} (%{repo_url})"),
-              filename: filename, repo_name: repo["name"], repo_url: repo["url"]
-            )
-          else
-            format(_("The file %{filename}"), filename: filename)
-          end
-
+          source = build_source(filename, repo_id)
           message = format(
-            _("%{source} is not digitally signed. The origin and integrity of the file cannot be " \
-              "verified. Use it anyway?"), source: source
+            _("%{source} is not digitally signed. The origin and integrity of the " \
```

Comment:
Please avoid building translated sentences from multiple parts. For translators it's difficult to translate the text correctly if they see only the beginning and cannot see the rest or vice versa. This usually results in a wrong or at least strange translation.

With translations, unlike with code, it's fine to have repeating or similar texts.

You can put the "The origin and integrity ..." part into a separate translatable string and merge them. That's fine because it is a full sentence.